src/files/handler.js

```diff
@@ -7,40 +8,45 @@ init();
  * Firebase Cloud Function handler for SvelteKit
  *
  * This function converts the Firebase Cloud Function (Express.js) Request object
- * into a format consumable to the SvelteKit render() function, which is of type
- * SvelteKit `import('types/hooks').StrictBody | null`
+ * into a format consumable to the SvelteKit render() function
  *
  * Relevant documentation - https://firebase.google.com/docs/functions/http-events#read_values_from_the_request
  *
  * @param {import('firebase-functions').https.Request} request
  * @param {import('express').Response} response
  * @returns {Promise<void>}
  */
-const svelteKit = async ({body, headers, method, rawBody, url}, response) => {
-	const host = `${headers['x-forwarded-proto']}://${headers.host}`;
-	const {pathname, searchParams: searchParameters = ''} = new URL(url || '', host);
+async function svelteKit(request, response) {
+	const rendered = await render(toSvelteKitRequest(request));
 
-	const finalRawBody =
-		headers['content-type'] === undefined ?
-			rawBody :
-			(headers['content-type'] === 'application/octet-stream' ?
-				body :
-				new TextDecoder(headers['content-encoding'] || 'utf-8').decode(rawBody));
+	return rendered ?
+		response.writeHead(rendered.status, rendered.headers).end(rendered.body) :
+		response.writeHead(404, 'Not Found').end();
+}
 
-	const rendered = await render({
-		method,
-		headers,
-		path: pathname,
-		query: searchParameters,
-		rawBody: finalRawBody
-	});
-
-	if (rendered) {
-		const {status, headers, body} = rendered;
-		return response.writeHead(status, headers).end(body);
-	}
+/**
+ * @param {import('firebase-functions').https.Request} request
+ * @return {import('@sveltejs/kit/types/internal').Incoming}
```

Comment:
Would be good if the types for the `init` and `render` functions were exported properly as I think this is bad form

Reply:
Yeah, you shouldn't have to reference anything from `internal`. PR welcome if you'd like to take a stab at it. CC @ignatiusmb who's been fixing up the types. Thanks to both of you for all the help!

Reply:
Started a discussion here - https://github.com/sveltejs/kit/issues/2249

Reply:
Cool. To answer @ignatiusmb's question, I haven't looked super closely, so am not 100% sure on this, but I feel like 90% sure that we just need to expose the types - though we should review and make sure it's all stuff we want to expose that would be my starting point